src/zcash/circuit/note.tcc

```diff
@@ -96,11 +96,6 @@ public:
         a_sk->generate_r1cs_constraints();
         rho->generate_r1cs_constraints();
 
-        // TODO: These constraints may not be necessary if SHA256
-        // already boolean constrains its outputs.
-        a_pk->generate_r1cs_constraints();
-        commitment->generate_r1cs_constraints();
-
```

Comment:
Add some comments:
```
// a_pk is boolean-constrained because it is the output of PRF^addr.
// commitment is boolean-constrained because it is the output of NoteCommitment.
```